Makefile

```diff
@@ -75,6 +75,14 @@ logs.%: ## Shows the logs of the given docker service. Example: make logs.datast
 restart.%: ## Restarts the given docker service. Example: make restart.datastore
 	docker compose restart $*
 
+run_tests.lints: ## Runs the linter tests
+	docker compose up dev-server -d
+	${SHELL_PREFIX} dev-server python3 -m scripts.linters.pre_commit_linter $(flags)
```

Comment:
Why do we need to run the dev-server to run pre-commit linter? Can't we run the backend-related lint checks on the backend image as a entry-point?

Reply:
the test needs python3.8 env that is within the backend dockerfile.

> Can't we run the backend-related lint checks on the backend image as a entry-point?
I am unable to understand running the lint checks as the entrypoint for backend image. what if dev wants to run the  checks manually sometime? like we run `python -m scripts.linters.pre_commit_linter` now

Reply:
they can run it manually similar to how we will run it in the CI, that should be fine, right?

Reply:
I have few question about this structure:

1. Why do we need to run Google app engine server for lint checks?
2. Why do we need to run webpack, angular-build container to run lint checks?

I hope this helps you understand my concerns.

Reply:
> they can run it manually similar to how we will run it in the CI, that should be fine, right?

yes.

> Why do we need to run Google app engine server for lint checks?

since we require python3.8 env to run tests, and we have python3.8 env in backend dockerfile + other packages we require in the python scripts are downloaded in the backend dockerfile + the volumes mapping for the oppia root dir in the dev-server enables lint checks to work properly.

> Why do we need to run webpack, angular-build container to run lint checks?

we don't require any server for the lint checks. its just we want to have python3.8 env and other packages required (present in the backend dockerfile)

Reply:
> since we require python3.8 env to run tests, and we have python3.8 env in backend dockerfile + other packages we require in the python scripts are downloaded in the backend dockerfile + the volumes mapping for the oppia root dir in the dev-server enables lint checks to work properly.

But it still doesn't explain why we need to run the server.

> we don't require any server for the lint checks. its just we want to have python3.8 env and other packages required (present in the backend dockerfile)

I understand you need python3.8 but it doesn't explain why we need to run so many services (7 or 8) to run lint checks.

Reply:
yes, we just need python3.8 env and other packages required which are present within the backend dockerfile, so we need to run a service that uses this backend dockerfile, and then run the command within that service. I choose dev-server service because we also need the volume mapping of oppia root dir or other generated files which dev-server have (datastore service do not have that vol mapping as it is not required for the datastore service).

yes, we don't need other services to be active for the lint checks, but since we start the devserver service, then all other services are started because dev-service depends on all other services

Reply:
> yes, we don't need other services to be active for the lint checks, but since we start the devserver service, then all other services are started because dev-service depends on all other services

It doesn't make sense to run 7 unwanted services to run lint checks, can we do it without running any unwanted service? Do you know any organization/project which follows this pattern? Is this the expected pattern to run the lint check with the docker project in other communities?